src/main/java/org/jenkinsci/plugins/workflow/job/WorkflowRun.java

```diff
@@ -565,11 +565,11 @@ private String key() {
                         }
 
                         if (Boolean.FALSE.equals(completed)) {
-                            // we've been restarted while we were running. let's get the execution going again.
-                            FlowExecutionListener.fireResumed(fetchedExecution);
-
                             getListener().getLogger().println("Resuming build at " + new Date() + " after Jenkins restart");
                             Timer.get().submit(() -> Queue.getInstance().schedule(new AfterRestartTask(WorkflowRun.this), 0)); // JENKINS-31614
+
+                            // we've been restarted while we were running. let's get the execution going again.
+                            FlowExecutionListener.fireResumed(fetchedExecution);
```

Comment:
Is this just so that the message gets printed first, or are there more significant effects?

Reply:
Mainly just to print the message first.